### ISSUE TITLE
MD013 "code_blocks => false" doesn't always work

### ISSUE THREAD

**Issue by markeganfuller** [markdownlint/markdownlint#141]
With the markdown:

<pre>
```bash
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
```

```bash
test

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
```                                                                                                                    

```bash
test
test

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
```
</pre>

And the style:

``` ruby
rule 'MD013', :code_blocks => false
```

I get the output:

``` bash
$ mdl --version
0.3.1
$ mdl -s test.rb test.md
test.md:15: MD013 Line length
```

Interestingly it works in the first two blocks but not the third.

**Comment by markeganfuller** [comment]
Looking deeper into this it seems the example isn't being detected as a line in a codeblock. 

Editing the test file `test/rule_tests/long_lines_code.md` to add this example and running the tests with a `puts codeblock_lines.join(",")` in the MD013 rule produces:

```
7,8,9,10,11,12,13,14,15
```

```
TestRules#test_long_lines_code [markdownlint/test/test_rules.rb:50]:
Expected: {"MD013"=>[3, 32]}
  Actual: {"MD013"=>[3, 16, 32]}
```

For some reason line 16 whilst being inside a codeblock is not detected by [this](https://github.com/mivok/markdownlint/blob/master/lib/mdl/rules.rb#L180):

``` ruby
# Every line in the document that is part of a code block.
codeblock_lines = doc.find_type_elements(:codeblock).map{
  |e| (doc.element_linenumber(e)..
       doc.element_linenumber(e) + e.value.count('\n') - 1).to_a }.flatten
```

Diffs for testing:

`````` diff
diff --git a/test/rule_tests/long_lines_code.md b/test/rule_tests/long_lines_code.md
index cf3d2f1..633249e 100644
--- a/test/rule_tests/long_lines_code.md
+++ b/test/rule_tests/long_lines_code.md
@@ -11,0 +12,7 @@ Here is a very very very very very very very very very very very very very very
+```text
+test
+test
+
+Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
+```
+
``````

``` diff
diff --git a/lib/mdl/rules.rb b/lib/mdl/rules.rb
index ed563b5..b4eddca 100644
--- a/lib/mdl/rules.rb
+++ b/lib/mdl/rules.rb
@@ -183,0 +184 @@ rule "MD013", "Line length" do
+    puts codeblock_lines.join(",")
```

**Comment by markeganfuller** [comment]
OK I think I have this that fixes it:

``` diff
diff --git a/lib/mdl/rules.rb b/lib/mdl/rules.rb
index ed563b5..d540526 100644
--- a/lib/mdl/rules.rb
+++ b/lib/mdl/rules.rb
@@ -180,7 +180,7 @@ rule "MD013", "Line length" do
     # Every line in the document that is part of a code block.
     codeblock_lines = doc.find_type_elements(:codeblock).map{
       |e| (doc.element_linenumber(e)..
-           doc.element_linenumber(e) + e.value.count('\n') - 1).to_a }.flatten
+           doc.element_linenumber(e) + e.value.lines.count).to_a }.flatten
     # Every line in the document that is part of a table.
     locations = doc.elements
                 .map { |e| [e.options[:location], e] }
```

Commit is [here](https://github.com/markeganfuller/markdownlint/commit/c5ccebe9da7cf1138bdb25263d7d201dd4b761ae) on my fork.

I added my example to the test and it seems to work ok, I'd like it if someone who actually knows ruby could check this before I do a PR.

Also if this is the correct fix, [this line](https://github.com/mivok/markdownlint/blob/master/lib/mdl/rules.rb#L167) might also need fixing.

**Comment by mivok** [comment]
Thanks for the thorough investigation into this. It looks like you're correct. The `'\n'` actually ends up counting the number of occurrences of a backslash or a letter n, and should have instead been `"\n"` in double quotes to match the newline. Changing it to `.lines.count` is cleaner though given that it's available.
